raft/node.go

```diff
@@ -484,5 +503,12 @@ func newReady(r *raft, prevSoftSt *SoftState, prevHardSt pb.HardState) Ready {
 	if r.raftLog.unstable.snapshot != nil {
 		rd.Snapshot = *r.raftLog.unstable.snapshot
 	}
+	if r.readIndex != None {
+		c := make([]byte, len(r.readIndexCtx))
+		copy(c, r.readIndexCtx)
```

Comment:
I'm not quite sure about how to deal with this, can we change readIndexCtx to requestID as an uint64 type, but we still need to convert it into []byte in raft.go since we need to put it into m.Entries

if keep the type as []byte, we have to copy it before returning it to the users so that it won't be changed by the next read index message